next/components/forms/useFormCurrentStepIndex.ts

```diff
@@ -1,71 +1,39 @@
-import { BAJSONSchema7 } from 'forms-shared/form-utils/ajvKeywords'
 import { createParser, useQueryState } from 'nuqs'
 import { useMemo, useRef } from 'react'
 import { useEffectOnce } from 'usehooks-ts'
 
-import {
-  getFirstNonEmptyStepIndex,
-  getStepProperty,
-  SUMMARY_HASH,
-} from '../../frontend/utils/formState'
-import { FormStepIndex } from './types/Steps'
-
-/**
- * Retrieves the query param of a step from provided schema.
- */
-const getStepQueryParam = (stepSchema: BAJSONSchema7 | null) => {
-  if (!stepSchema) {
-    return null
-  }
-
-  const stepProperty = getStepProperty(stepSchema)
-  if (!stepProperty) {
-    return null
-  }
-
-  const step = stepSchema?.properties?.[stepProperty] as BAJSONSchema7 & { hash?: string }
-  return step.hash
-}
+import { FormStepIndex, FormStepperStep } from './types/Steps'
 
 const getStepIndexByQueryParam = (
-  stepSchemas: (BAJSONSchema7 | null)[],
+  steps: FormStepperStep[],
   queryParam: string | null | undefined,
 ) => {
-  if (!queryParam) {
+  if (queryParam == null) {
```

Comment:
We had some discussions in PAAS and OLO projects about `== null` and agreed on using `!isDefined()` instead, for better readability (also for more junior people). Even though it's normal pattern to use double `=` in this case (comparing agains `null` or `undefined`), but not very easy to distinguish for juniors.